strings/base_coroutine_foundation.h

```diff
@@ -801,4 +801,43 @@ WINRT_EXPORT namespace winrt
         impl::check_status_canceled(shared->status);
         co_return shared->result.GetResults();
     }
+
+    template <typename Delegate>
+    struct async_event : impl::event_base<Delegate>
+    {
+        async_event() = default;
+
+        template<typename...Arg>
+        Windows::Foundation::IAsyncAction operator()(Arg const... args)
```

Comment:
There's also a usability issue here. Because the args are necessarily `const` rather than `const&`, the callsite cannot simply use `m_event(*this, args)` but must explicitly cast as in `m_event(static_cast<SenderType>(*this), args)`.

Reply:
(sad face)

Reply:
Another approach is to have `async_event`'s `operator()` be a blocking call that runs the handlers in parallel. If the caller wants to avoid blocking the call itself they can simply run it on a background thread like you can with `event`. That would solve both this usability problem and the issue of removing dead handlers.

Reply:
Maybe at the end of the day we don't need this at all. Callers can do

```cpp
[](auto lifetime, auto args) -> winrt::fire_and_forget
{
    winrt::resume_background();
    lifetime->m_event(*lifetime, args);
}(get_strong(), args);
```

Reply:
Right, but that would still allow one long-running handler to delay other handlers from running. I just don't know whether that's a scenario anyone cares about.